#' chm13v2.1000kbp.PE100
#'
#' Bin annotations for 1000kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.1000kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 1000kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.1000kbp.PE100"


#' chm13v2.1000kbp.SR50
#'
#' Bin annotations for 1000kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.1000kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 1000kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.1000kbp.SR50"


#' chm13v2.100kbp.PE100
#'
#' Bin annotations for 100kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.100kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 100kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.100kbp.PE100"


#' chm13v2.100kbp.SR50
#'
#' Bin annotations for 100kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.100kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 100kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.100kbp.SR50"


#' chm13v2.10kbp.PE100
#'
#' Bin annotations for 10kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.10kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 10kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.10kbp.PE100"


#' chm13v2.10kbp.SR50
#'
#' Bin annotations for 10kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.10kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 10kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.10kbp.SR50"


#' chm13v2.150kbp.PE100
#'
#' Bin annotations for 150kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.150kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 150kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.150kbp.PE100"


#' chm13v2.150kbp.SR50
#'
#' Bin annotations for 150kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.150kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 150kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.150kbp.SR50"


#' chm13v2.15kbp.PE100
#'
#' Bin annotations for 15kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.15kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 15kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.15kbp.PE100"


#' chm13v2.15kbp.SR50
#'
#' Bin annotations for 15kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.15kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 15kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.15kbp.SR50"


#' chm13v2.1kbp.PE100
#'
#' Bin annotations for 1kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.1kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 1kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.1kbp.PE100"


#' chm13v2.1kbp.SR50
#'
#' Bin annotations for 1kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.1kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 1kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.1kbp.SR50"


#' chm13v2.200kbp.PE100
#'
#' Bin annotations for 200kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.200kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 200kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.200kbp.PE100"


#' chm13v2.200kbp.SR50
#'
#' Bin annotations for 200kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.200kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 200kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.200kbp.SR50"


#' chm13v2.20kbp.PE100
#'
#' Bin annotations for 20kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.20kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 20kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.20kbp.PE100"


#' chm13v2.20kbp.SR50
#'
#' Bin annotations for 20kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.20kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 20kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.20kbp.SR50"


#' chm13v2.30kbp.PE100
#'
#' Bin annotations for 30kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.30kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 30kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.30kbp.PE100"


#' chm13v2.30kbp.SR50
#'
#' Bin annotations for 30kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.30kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 30kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.30kbp.SR50"


#' chm13v2.500kbp.PE100
#'
#' Bin annotations for 500kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.500kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 500kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.500kbp.PE100"


#' chm13v2.500kbp.SR50
#'
#' Bin annotations for 500kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.500kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 500kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.500kbp.SR50"


#' chm13v2.50kbp.PE100
#'
#' Bin annotations for 50kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.50kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 50kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.50kbp.PE100"


#' chm13v2.50kbp.SR50
#'
#' Bin annotations for 50kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.50kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 50kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.50kbp.SR50"


#' chm13v2.5kbp.PE100
#'
#' Bin annotations for 5kbpkbp using PE100 reads
#'
#' @docType data
#' @usage data(chm13v2.5kbp.PE100)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 5kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.5kbp.PE100"


#' chm13v2.5kbp.SR50
#'
#' Bin annotations for 5kbpkbp using SR50 reads
#'
#' @docType data
#' @usage data(chm13v2.5kbp.SR50)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{chromosome}{Chromosome name}
#'   \item{start}{Bin start position}
#'   \item{end}{Binend position}
#'   \item{bases}{Percentage of non-N nucleotides acros the full bin}
#'   \item{gc}{Percentage of C and G nucleotides \(of non-N nucleotides\)}
#'   \item{mappability}{Average mappability of 5kbp with a maximum of 2 mismatches}
#'   \item{blacklist}{Percent overlap with High-Signal regions from ExcludeRanges}
#'   \item{residual}{Median loess residual from 1000 genomes **PENDING**}
#'   \item{use}{Weather bin exceeds minimum thresholds for analysis}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
#' @import QDNAseq
#' @importFrom Biobase AnnotatedDataFrame
"chm13v2.5kbp.SR50"


#' used.fastq.manifest
#'
#' Manifest of FASTQ sequences used to estimate residuals
#'
#' @docType data
#' @usage data(used.fastq.manifest)
#' @format A annotated data frame with pre-calcuated metrics for each bin describing:
#' \describe{
#'   \item{url}{FASTQ download URL}
#'   \item{md5}{MD5 sums}
#'   \item{Data.collection}{Data Collection at 1000 Genomes Project}
#'   \item{Data.type}{type of data}
#'   \item{Analysis.group}{PCR-free high coverage}
#'   \item{Sample}{Sample Name}
#'   \item{Population}{Population represented in the 1000 Genomes Project}
#' }
#'
#' @author RGM
#' @source \url{https://github.com/RodrigoGM/QDNAseq.chm13v2}
"used.fastq.manifest"


